openedx/core/lib/xblock_builtin/xblock_discussion/tests.py

```diff
@@ -0,0 +1,153 @@
+""" Tests for DiscussionXBLock"""
+import ddt
+import mock
+from nose.plugins.attrib import attr
+from unittest import TestCase
+
+from safe_lxml import etree
+
+from openedx.core.lib.xblock_builtin.xblock_discussion.xblock_discussion import DiscussionXBlock
+from xblock.field_data import DictFieldData
+from xblock.fields import ScopeIds, UNIQUE_ID, NO_CACHE_VALUE
+from xblock.runtime import Runtime
+
+
+@attr('shard2')
+@ddt.ddt
+class DiscussionXBlockImportExportTests(TestCase):
+    """
+    Import and export tests
+    """
+    DISCUSSION_XBLOCK_LOCATION = "openedx.core.lib.xblock_builtin.xblock_discussion.xblock_discussion.DiscussionXBlock"
+
+    def setUp(self):
+        """
+        Set up method
+        """
+        super(DiscussionXBlockImportExportTests, self).setUp()
+        self.keys = ScopeIds("any_user", "discussion", "def_id", "usage_id")
+        self.runtime_mock = mock.Mock(spec=Runtime)
+        self.runtime_mock.construct_xblock_from_class = mock.Mock(side_effect=self._construct_xblock_mock)
+        self.runtime_mock.get_policy = mock.Mock(return_value={})
+        self.id_gen_mock = mock.Mock()
+
+    def _construct_xblock_mock(self, cls, keys):  # pylint: disable=unused-argument
+        """
+        Builds target xblock instance (DiscussionXBlock)
+
+        Signature-compatible with runtime.construct_xblock_from_class - can be used as a mock side-effect
+        """
+        return DiscussionXBlock(self.runtime_mock, scope_ids=keys, field_data=DictFieldData({}))
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("Test category", "Test target"),
+        ("Some other category", "Some other target")
+    )
+    def test_xblock_export_format(self, category, target, patched_load_definition_xml):
+        """
+        Test that xblock export XML format can be parsed preserving field values
+        """
+        xblock_xml = """
+        <discussion
+            url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" xblock-family="xblock.v1"
+            discussion_category="{category}"
+            discussion_target="{target}"
```

Comment:
@cpennington With the old XModule, if I import 
```xml
<discussion id="custom_id_1" discussion_category="Custom Topic" for="Custom sub-topic" display_name="Externalized Discussion with Custom Topic"/>
```
and then export it immediately, I get
```xml
<discussion id="custom_id_1" for="Custom sub-topic" discussion_category="Custom Topic" discussion_id="custom_id_1" discussion_target="Custom sub-topic" display_name="Externalized Discussion with Custom Topic"/>
```

If I import 
```xml
<discussion discussion_id="custom_id_4" discussion_category="Custom Topic 2" discussion_target="Custom sub-topic 2" display_name="Externalized Discussion with Custom Topic 2"/>
```
and export I get 
```xml
<discussion discussion_category="Custom Topic 2" discussion_id="custom_id_4" discussion_target="Custom sub-topic 2" display_name="Externalized Discussion with Custom Topic 2"/>

```
(the same)

So that seems to indicate that the export should use `discussion_id` and `discussion_target` but needs to be able to import `id` and `for`. I think that is the case with Eugeny's latest change, and am just trying to test comprehensively.